br: volume-snapshot backup and restore failure when backup cluster is empty and there are only one region

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

1. create a cluster with tidb configure split-table=false
2. wait and check region by pd-ctl, ensure only one region existed in cluster
3. volume snapshot backup the cluster (only one region with startKey="" and endKey="")
4. create a restore cluster with recoveryMode: true
5. create a volume-snapshot restore

for details refer to 
backup https://docs.pingcap.com/zh/tidb-in-kubernetes/stable/backup-to-aws-s3-by-snapshot
restore https://docs.pingcap.com/zh/tidb-in-kubernetes/stable/restore-from-aws-s3-by-snapshot
### 2. What did you expect to see? (Required)
restore shall success
### 3. What did you see instead (Required)
the 2nd restore failure with error
`I0205 02:25:43.331325       9 restore.go:164] [2023/02/05 02:25:43.330 +00:00] [ERROR] [restore.go:51] ["failed to restore data"] [error="receive Regions with no peer"] [errorVerbose="receive Regions with no peer\ngithub.com/tikv/cli
ent-go/v2/internal/locate.(*RegionCache).scanRegions\n\t/home/ophone/dev/client-go/internal/locate/region_cache.go:1605\ngithub.com/tikv/client-go/v2/internal/locate.(*RegionCache).BatchLoadRegionsWithKeyRange\n\t/home/ophone/dev/cli
ent-go/internal/locate/region_cache.go:1194\ngithub.com/tikv/client-go/v2/internal/locate.(*RegionCache).BatchLoadRegionsFromKey\n\t/home/ophone/dev/client-go/internal/locate/region_cache.go:1219\ngithub.com/tikv/client-go/v2/txnkv/r
angetask.(*Runner).RunOnRange\n\t/home/ophone/dev/client-go/txnkv/rangetask/range_task.go:191\ngithub.com/pingcap/tidb/br/pkg/restore.(*Recovery).PrepareFlashbackToVersion.func1\n\t/home/ophone/dev/tidb/br/pkg/restore/data.go:315\ngi
thub.com/pingcap/tidb/br/pkg/utils.WithRetry\n\t/home/ophone/dev/tidb/br/pkg/utils/retry.go:52\ngithub.com/pingcap/tidb/br/pkg/restore.(*Recovery).PrepareFlashbackToVersion\n\t/home/ophone/dev/tidb/br/pkg/restore/data.go:305`
### 4. What is your TiDB version? (Required)
v6.5.0 GA and late
<!-- Paste the output of SELECT tidb_version() -->

--- a/br/pkg/restore/data.go
+++ b/br/pkg/restore/data.go
@@ -302,22 +302,29 @@ func (recovery *Recovery) WaitApply(ctx context.Context) (err error) {
 
 // prepare the region for flashback the data, the purpose is to stop region service, put region in flashback state
 func (recovery *Recovery) PrepareFlashbackToVersion(ctx context.Context, resolveTS uint64, startTS uint64) (err error) {
-	handler := func(ctx context.Context, r tikvstore.KeyRange) (rangetask.TaskStat, error) {
-		stats, err := ddl.SendPrepareFlashbackToVersionRPC(ctx, recovery.mgr.GetStorage().(tikv.Storage), resolveTS, startTS, r)
-		return stats, err
-	}
+	retryErr := utils.WithRetry(
+		ctx,
+		func() error {
+			handler := func(ctx context.Context, r tikvstore.KeyRange) (rangetask.TaskStat, error) {
+				stats, err := ddl.SendPrepareFlashbackToVersionRPC(ctx, recovery.mgr.GetStorage().(tikv.Storage), resolveTS, startTS, r)
+				return stats, err
+			}
 
-	runner := rangetask.NewRangeTaskRunner("br-flashback-prepare-runner", recovery.mgr.GetStorage().(tikv.Storage), int(recovery.concurrency), handler)
-	// Run prepare flashback on the entire TiKV cluster. Empty keys means the range is unbounded.
-	err = runner.RunOnRange(ctx, []byte(""), []byte(""))
-	if err != nil {
-		log.Error("region flashback prepare get error")
-		return errors.Trace(err)
-	}
+			runner := rangetask.NewRangeTaskRunner("br-flashback-prepare-runner", recovery.mgr.GetStorage().(tikv.Storage), int(recovery.concurrency), handler)
+			// Run prepare flashback on the entire TiKV cluster. Empty keys means the range is unbounded.
+			err = runner.RunOnRange(ctx, []byte(""), []byte(""))
+			if err != nil {
+				log.Warn("region flashback prepare get error")
+				return errors.Trace(err)
+			}
+			log.Info("region flashback prepare complete", zap.Int("regions", runner.CompletedRegions()))
+			return nil
+		},
+		utils.NewFlashBackBackoffer(),
+	)
+
 	recovery.progress.Inc()
-	log.Info("region flashback prepare complete", zap.Int("regions", runner.CompletedRegions()))
-
-	return nil
+	return retryErr
 }
 
 // flashback the region data to version resolveTS

--- a/br/pkg/utils/backoff.go
+++ b/br/pkg/utils/backoff.go
@@ -33,6 +33,11 @@ const (
 	resetTSRetryTimeExt       = 600
 	resetTSWaitIntervalExt    = 500 * time.Millisecond
 	resetTSMaxWaitIntervalExt = 300 * time.Second
+
+	// region heartbeat are 10 seconds by default, if some region has 2 heartbeat missing (15 seconds), it appear to be a network issue between PD and TiKV.
+	flashbackRetryTime       = 3
+	flashbackWaitInterval    = 3000 * time.Millisecond
+	flashbackMaxWaitInterval = 15 * time.Second
 )
 
 // RetryState is the mutable state needed for retrying.
@@ -202,5 +207,36 @@ func (bo *pdReqBackoffer) NextBackoff(err error) time.Duration {
 }
 
 func (bo *pdReqBackoffer) Attempt() int {
+	return bo.attempt
+}
+
+type flashbackBackoffer struct {
+	attempt      int
+	delayTime    time.Duration
+	maxDelayTime time.Duration
+}
+
+// NewBackoffer creates a new controller regulating a truncated exponential backoff.
+func NewFlashBackBackoffer() Backoffer {
+	return &flashbackBackoffer{
+		attempt:      flashbackRetryTime,
+		delayTime:    flashbackWaitInterval,
+		maxDelayTime: flashbackMaxWaitInterval,
+	}
+}
+
+// retry 3 times when prepare flashback failure.
+func (bo *flashbackBackoffer) NextBackoff(err error) time.Duration {
+	bo.delayTime = 2 * bo.delayTime
+	bo.attempt--
+	log.Warn("region may not ready to serve, retry it...", zap.Error(err))
+
+	if bo.delayTime > bo.maxDelayTime {
+		return bo.maxDelayTime
+	}
+	return bo.delayTime
+}
+
+func (bo *flashbackBackoffer) Attempt() int {
 	return bo.attempt
 }

